proofs/tree.go

```diff
@@ -129,7 +129,7 @@ package proofs
 import (
 	"bytes"
 	"crypto/rand"
-	"encoding/base64"
+	"encoding/hex"
```

Comment:
Would you like to use this package instead - https://godoc.org/github.com/ethereum/go-ethereum/common/hexutil

This would keep us consistent across. Else we have to prepend '0x'

Reply:
I am not sure about that, `precise-proofs` should not have a dependency on ethereum library. This will force users to import a non standard library (ethereum/go-ethereum) into their projects.
As for us, proof generation doesn't need to know formats, just field names and proof validation on front end works with and without `0x` prefixed.

Reply:
I feel that the library must not dictate which format the user must use to encode binary to string. Can we make the `EncodeToString` an interface and ask the user to pass in the encoder?

Reply:
I wouldn't consider making a very common functionality such as serialisation pluggable over-engineering. Do you think it's a major effort to add/use an interface here?

Reply:
I consider over-engineer when there is no use case to have that as pluggable and we spend time refactoring the code where we can use that effort on something more urgent.
Disregarding the effort to implement that change I agree that it will look more sophisticated but I rather keep it as simple as possible for now.

Reply:
the use case is already here, as we needed to switch the encoding format from base64 to hex IMO.

Reply:
@vimukthi-git we are switching the implementation not supporting multiple types of encoders, because we can make that decision right now. Please contact me privately if you need to discuss anything further.

Reply:
@mikiquantum if you think adding another dependency is unnecessary, maybe a helper function to prepend '0x' works?

Reply:
The main limitation of prepending the "0x" to the hex string is that we are forcing the client to use a custom decoder (made by us or them) to get the value  into hex.